src/main/java/org/cryptomator/cli/CryptomatorCli.java

```diff
@@ -56,38 +58,73 @@ private static void validate(Args args) throws IllegalArgumentException {
 				throw new IllegalArgumentException("Not a directory: " + vaultPath);
 			}
 			args.addPasswortStrategy(vaultName).validate();
+
+			Path mountPoint = args.getFuseMountPoint(vaultName);
+			if (mountPoint != null && !Files.isDirectory(mountPoint)) {
+				throw new IllegalArgumentException("Fuse mount point does not exist: " + mountPoint);
+			}
 		}
 	}
 
 	private static void startup(Args args) throws IOException {
-		WebDavServer server = WebDavServer.create();
-		server.bind(args.getBindAddr(), args.getPort());
-		server.start();
+		WebDav server = initWebDavServer(args);
+		ArrayList<FuseMount> mounts = new ArrayList<>();
 
 		for (String vaultName : args.getVaultNames()) {
 			Path vaultPath = Paths.get(args.getVaultPath(vaultName));
 			LOG.info("Unlocking vault \"{}\" located at {}", vaultName, vaultPath);
 			String vaultPassword = args.getPasswordStrategy(vaultName).password();
-			CryptoFileSystemProperties properties = CryptoFileSystemProperties.cryptoFileSystemProperties().withPassphrase(vaultPassword).build();
+			CryptoFileSystemProperties properties = CryptoFileSystemProperties.cryptoFileSystemProperties()
+					.withPassphrase(vaultPassword).build();
 			Path vaultRoot = CryptoFileSystemProvider.newFileSystem(vaultPath, properties).getPath("/");
-			WebDavServletController servlet = server.createWebDavServlet(vaultRoot, vaultName);
-			servlet.start();
+
+			Path fuseMountPoint = args.getFuseMountPoint(vaultName);
+			if (fuseMountPoint != null) {
+				FuseMount newMount = new FuseMount(vaultRoot, fuseMountPoint);
+				if (newMount.mount()) {
+					mounts.add(newMount);
+				}
+			}
+
+			if (server != null) {
+				server.addServlet(vaultRoot, vaultName);
+			}
 		}
 
 		waitForShutdown(() -> {
 			LOG.info("Shutting down...");
 			try {
-				server.stop();
+				if (server != null) {
+					server.stop();
+				}
+
+				for (FuseMount mount : mounts) {
+					mount.umount();
+				}
 				LOG.info("Shutdown successful.");
 			} catch (Throwable e) {
 				LOG.error("Error during shutdown", e);
 			}
 		});
 	}
 
+	private static WebDav initWebDavServer(Args args) {
+		WebDav server = null;
+		if (args.hasValidWebDavConf()) {
+			server = new WebDav(args.getBindAddr(), args.getPort());
+		}
+		return server;
+	}
+
 	private static void waitForShutdown(Runnable runnable) {
 		Runtime.getRuntime().addShutdownHook(new Thread(runnable));
-		LOG.info("Server started. Press Ctrl+C to terminate.");
+		LOG.info("Press Ctrl+C to terminate.");
+		try {
+			while (true) {
+				System.in.read();
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
```

Comment:
why is this?

Reply:
> Please let me know if you have suggestions how to achieve this more cleanly!

You can run FUSE in "blocking mode" (second argument of the `mount` method), but you'd need to also overload the Mounter's mount method, as blocking is currently always `false`:

https://github.com/cryptomator/fuse-nio-adapter/blob/develop/src/main/java/org/cryptomator/frontend/fuse/mount/LinuxMounter.java#L22

I'd prefer this approach but leave it open to you.

Reply:
Thanks for the suggestion. I have a couple of concerns, though. Please let me know if they are valid:
- If I mount in blocking mode it implies that I can only have a single FUSE mount when running the CLI-app, right? Or surely I could add some threading, but that would IMO complicate this app unnecessarily. With the current approach there can be multiple simultaneous FUSE and/or WebDAV mounts running with a single CLI-app instance and the logic is fairly straight forward.
- What is the release cadence of fuse-nio-adapter? I mean if I make a pull request, when will it be available in Maven?
- I probably should also make similar changes to MacMounter.java. I currently do not have means to test that it works.

Reply:
> If I mount in blocking mode it implies that I can only have a single FUSE mount when running the CLI-app, right? Or surely I could add some threading, but that would IMO complicate this app unnecessarily. With the current approach there can be multiple simultaneous FUSE and/or WebDAV mounts running with a single CLI-app instance and the logic is fairly straight forward.

You're right. You can only have a single mount in this case. Ctrl+C will unmount it and end the process.

> What is the release cadence of fuse-nio-adapter? I mean if I make a pull request, when will it be available in Maven?

It can be released any time, there is no schedule we need to follow. It is just a matter of tagging and letting CI do its job.

> I probably should also make similar changes to MacMounter.java. I currently do not have means to test that it works.

MacMounter has the same boolean flag, which works the same way. I have already tested it locally.

Reply:
Ok. I think I would then keep this functionality as it is currently implemented as the previous WebDAV implementation also supports multiple mounts. Now you can basically add any number of WebDAV and/or FUSE mounts with a single instance.

From my side this PR is ready to be merged now in case it's fine for you too.

Reply:
> Now you can basically add any number of WebDAV and/or FUSE mounts with a single instance.

I still think this is a debatable feature, since you can as well spawn multiple processes and thus have better control over each mounted vault. However, this is out of scope of this PR and should be discussed in a separate issue.

Reply:
Might this cause #35?